airflow/models/abstractoperator.py

```diff
@@ -149,6 +155,92 @@ def dag_id(self) -> str:
     def node_id(self) -> str:
         return self.task_id
 
+    @property
+    def is_setup(self):
+        """
+        Whether the operator is a setup task.
+
+        :meta private:
+        """
+        return self._is_setup
+
+    @is_setup.setter
+    def is_setup(self, value):
+        """
+        Setter for is_setup property.
+
+        :meta private:
+        """
+        if self.is_teardown is True and value is True:
```

Comment:
```suggestion
        if self.is_teardown is True:
```
I think being strict here is better to avoid bad coding from 'copying'

Reply:
oh i think i see what you mean

Reply:
ok i tried it but yeah we can really do this becasue it then it prevents setting it with even the default value at init.  reverted for now. if you have another suggestion then i'm all ears